core/src/main/resources/reference.conf

```diff
@@ -420,6 +420,18 @@ datastax-java-driver {
     #   change.
     # Overridable in a profile: no
     warn-on-init-error = true
+
+    # The timeout to use when establishing driver connections.
+    #
+    # This timeout is for controlling how long the driver will wait for the underlying channel
+    # to actually connect to the server. This is not the time limit for completing protocol
+    # negotiations, only the time limit for establishing a channel connection.
+    #
+    # Required: yes
+    # Modifiable at runtime: yes, the new value will be used for connections created after the
+    #   change.
+    # Overridable in a profile: no
+    connect-timeout = 5 seconds
```

Comment:
why `5 seconds`?

Reply:
5 seconds was the default for the 3.x driver: https://github.com/datastax/java-driver/blob/3.x/driver-core/src/main/java/com/datastax/driver/core/SocketOptions.java#L23-L29

Reply:
Shouldn't we increase that? In driver 3 timeouts were more fine-grain. In 4.x the request timeout is taking whole path of request into account - so I suspect it is higher. In such a case maybe connect timeout should be also higher?

Reply:
This is just the Netty channel connection timeout though. I don't think it affects or is affected by request processing. Just how long Netty will wait before giving up trying to establish an I/O connection to the server.

Reply:
👍 
This is unrelated to `basic.request.timeout`. It will only affect session initialization, or background reconnections. The same default as 3.x is fine.